### PR TITLE
Client: Remove http client

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following removes the http client, we should no longer be using this
and instead, we should use juju/http.Client.

This formalizes the process.

### CODE REVIEW

**Comment by hmlanigan** [comment]
Very nice!

I'll look after the build failure is resolved.  We should rev utils for this, jik.

**Comment by SimonRichardson** [comment]
Yeah true that

**Comment by SimonRichardson** [comment]
!!build!!

**Review by hmlanigan — COMMENTED** [review]
LGTM

File: cert/cert_test.go [hunk 1]
```diff
@@ -125,7 +124,10 @@ func checkTLSConnection(c *gc.C, caCert, srvCert *x509.Certificate, srvKey *rsa.
 	var clientState tls.ConnectionState
 	done := make(chan error)
 	go func() {
-		config := utils.SecureTLSConfig()
+		config := &tls.Config{
+			CipherSuites: knownGoodCipherSuites,
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply cert/cert_test.go:128]: I'm seriously considering bringing juju/http as a dependency to solve this

[the remark this answers is not in the document]
Comment (SimonRichardson) [reply cert/cert_test.go]: Turns out I could just delete a ton of code.

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Review by hmlanigan — APPROVED** [review]

**Review by hmlanigan — APPROVED** [review]

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

The ssh tests are broken and intermediately passing.

**Pull request merged**